src/components/Network.vue

```diff
@@ -0,0 +1,216 @@
+                ).serialize(),
+                keyguardRequest.networkId,
+            );
+        } else {
+            tx = new Nimiq.BasicTransaction(
+                new Nimiq.PublicKey(keyguardResult.publicKey),
+                new Nimiq.Address(new Nimiq.SerialBuffer(keyguardRequest.recipient)),
+                keyguardRequest.value,
+                keyguardRequest.fee,
+                keyguardRequest.validityStartHeight,
+                new Nimiq.Signature(keyguardResult.signature),
+                keyguardRequest.networkId,
+            );
+        }
+
+        return tx;
+    }
+
+    /**
+     * The result of this method can also be used as a PlainTransaction
+     * for all relevant nano-api methods.
+     */
+    public makeSignTransactionResult(tx: Nimiq.Transaction): SignTransactionResult {
+        const proof = Nimiq.SignatureProof.unserialize(new Nimiq.SerialBuffer(tx.proof));
+
+        const result: SignTransactionResult = {
+            serializedTx: tx.serialize(),
+
+            sender: tx.sender.toUserFriendlyAddress(),
+            senderType: tx.senderType,
+            senderPubKey: proof.publicKey.serialize(),
+
+            recipient: tx.recipient.toUserFriendlyAddress(),
+            recipientType: tx.recipientType,
+
+            value: Nimiq.Policy.satoshisToCoins(tx.value),
+            fee: Nimiq.Policy.satoshisToCoins(tx.fee),
+            validityStartHeight: tx.validityStartHeight,
+
+            signature: proof.signature.serialize(),
+
+            extraData: tx.data,
+            flags: tx.flags,
+            networkId: tx.networkId,
+
+            hash: tx.hash().toBase64(),
+        };
+
+        return result;
+    }
+
+    /**
+     * Relays the transaction to the network and only returns when the network
+     * fires its 'transaction-relayed' event for that transaction.
+     */
+    public async sendToNetwork(tx: Nimiq.Transaction): Promise<SignTransactionResult> {
+        const txObj = this.makeSignTransactionResult(tx);
+        const client = await this._getNetworkClient();
+
+        client.relayTransaction(txObj);
+
+        return new Promise<SignTransactionResult>((resolve, reject) => {
+            this.$once('transaction-relayed', (txInfo: any) => {
+                if (txInfo.hash === txObj.hash) resolve(txObj);
+            });
+        });
+    }
+
+    public async getBalances(addresses: string[]): Promise<Map<string, number>> {
+        const client = await this._getNetworkClient();
+        return client.getBalance(addresses);
+    }
+
+    private created() {
+        this.$on('consensus-established', () => this.consensusEstablished = true);
+        this.$on('consensus-lost', () => this.consensusEstablished = false);
+    }
+
+    private async _getNetworkClient(): Promise<NetworkClient> {
+        if (this._networkClient) return this._networkClient;
+
+        this._networkClient = new NetworkClient('http://localhost:5000');
```

Comment:
This needs to be adapted to the respective development environment, or we define a standard where we always start our network locally. I think we cannot use the deployed network from our local environment, because the deployment uses HTTPS which we cannot communicate to and from from HTTP localhost.